lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -415,13 +415,63 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
 // VerilogEmitter
 //===----------------------------------------------------------------------===//
 
+void VerilogEmitterOptions::parse(StringRef text, ErrorHandlerT errorHandler) {
+  while (!text.empty()) {
+    // Remove the first option from the text.
+    auto split = text.split(",");
+    auto option = split.first.trim();
+    text = split.second;
+    if (option == "") {
+      // Empty options are fine.
+    } else if (option == "noAlwaysFF") {
+      useAlwaysFF = false;
+    } else {
+      errorHandler(llvm::Twine("unknown style option \'") + option + "\'");
+      // We continue parsing options after a failure.
+    }
+  }
+}
+
+std::string VerilogEmitterOptions::toString() const {
+  std::string options = "";
+  if (!useAlwaysFF)
+    options += "noAlwaysFF";
+  return options;
+}
+
+void VerilogEmitterOptions::print(raw_ostream &os) const { os << toString(); }
+
+/// Read the verilog emitter options out of a module and into an options struct.
+/// If there are no options specified on the module, then nothing is changed in
+/// the options struct.
+static void readVerilogEmitterOptions(ModuleOp module,
+                                      VerilogEmitterOptions &options) {
+  // Load any emitter options from the module.
+  if (auto styleAttr = module->getAttrOfType<StringAttr>("circt.outputStyle")) {
+    options.parse(styleAttr.getValue(),
+                  [&](Twine error) { module.emitError(error); });
+  }
+}
+
+/// Write the verilog emitter options from an option struct and in to a module's
+/// attributes.
+static void writeVerilogEmitterOptions(ModuleOp module,
+                                       const VerilogEmitterOptions &options) {
+  module->setAttr("circt.outputStyle",
+                  StringAttr::get(module.getContext(), options.toString()));
```

Comment:
similarly this can be a `optionset.setAsAttribute(someModule)` method or something